src/compiler/checker.ts

```diff
@@ -10403,6 +10433,7 @@ namespace ts {
                 case SyntaxKind.TemplateExpression:
                     return checkTemplateExpression(<TemplateExpression>node);
                 case SyntaxKind.StringLiteral:
+                    return checkStringLiteralExpression(<LiteralExpression>node);
                 case SyntaxKind.NoSubstitutionTemplateLiteral:

```

Comment:
@DanielRosenwasser It's better but lets take a concrete example in ```lib.d.ts```:
``` 
addEventListener(type: "MSPointerCancel", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
addEventListener(type: "MSPointerDown", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
addEventListener(type: "MSPointerEnter", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
addEventListener(type: "MSPointerLeave", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
addEventListener(type: "MSPointerMove", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
addEventListener(type: "MSPointerOut", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
addEventListener(type: "MSPointerOver", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
addEventListener(type: "MSPointerUp", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void; 
```
You can optimize that by reducing to a ```single``` function (for overload checking) and a ```single``` type:
```
addEventListener(type: "MSPointerCancel" | "MSPointerDown" ... | "MSPointerUp", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;

addEventListener("MSPointerUp"); // boom done
 // constant check of "MSPointerUp" in the StringSet type with values "MSPointerCancel" | "MSPointerDown" | ... | "MSPointerUp"
```
It's probably a good idea to do that reduction too with your union of string literals.

I mean take an extreme example where someone could union 100 string literals:
```
addEventListener(type: "MSPointerCancel" | ... | 100 of these | "end", listener: (ev: MSPointerEvent) => any, useCapture?: boolean): void;
``` 
addEventListener("100 string checks will be slow in a loop");